doc/man7/SM2.pod

```diff
@@ -0,0 +1,77 @@
+=pod
+
+=head1 NAME
+
+SM2 - Chinese SM2 signature and encryption algorithm support
+
+=head1 DESCRIPTION
+
+B<SM2> algorithm is first defined by the Chinese national standard GM/T 0003-2012
+and is standardized by ISO as ISO/IEC 14888. B<SM2> is actually an elliptic curve
+based algorithm. The current implementation in OpenSSL supports both signature and
+encryption schemes via EVP interface.
+
+When doing the B<SM2> signature algorithm, it requires a distinguishing identifier
+to form the message prefix which is hashed before the real message is hashed.
+
+=head1 NOTES
+
+B<SM2> signature can be generated by using the 'DigestSign' series APIs, for instance,
+EVP_DigestSignInit(), EVP_DigestSignUpdate() and EVP_DigestSignFinal(). Ditto
+for the verification process by calling the 'DigestVerify' series APIs.
+
+There are several special steps need to be done before computing an B<SM2> signature.
+
+The B<EVP_PKEY> structure should be set to B<EVP_PKEY_SM2> by calling:
+
+ EVP_PKEY_set_alias_type(pkey, EVP_PKEY_SM2);
+
+Then an ID should be set by calling:
+
+ EVP_PKEY_CTX_set1_id(pctx, id, id_len);
+
+When calling the EVP_DeigestSignInit() or EVP_DigestVerifyInit() function, a
+pre-allocated B<EVP_PKEY_CTX> should be assigned to the B<EVP_MD_CTX>. This is
+done by calling:
+
+ EVP_MD_CTX_set_pkey_ctx(mctx, pctx);
+
+And normally there is no need to pass a B<pctx> parameter to EVP_DeigestSignInit()
+or EVP_DigestVerifyInit() in such scenario.
+
+=head1 EXAMPLE
+
+This example demonstrates the calling sequence on how to use an B<EVP_PKEY> to
+sign a message with SM2 signature algorithm and SM3 hash algorithm:
```

Comment:
with *the* SM2....and *the* SM3

Reply:
Perhaps add this sentence. "The example below omits error checking code for clarity. Application code should check the return codes of function calls for errors."

Reply:
The text above says this is an example for signing. But actually its an example for verifying a signature.